app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.kt

```diff
@@ -0,0 +1,29 @@
+package ar.com.wolox.android.example.ui.login
+
+import android.util.Patterns
+import ar.com.wolox.android.example.utils.UserSession
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import javax.inject.Inject
+
+class LoginPresenter @Inject constructor(private val userSession: UserSession) : BasePresenter<LoginView>() {
+
+    fun onLoginButtonClicked(user: String, password: String) {
+        if (user.isNotEmpty() && password.isNotEmpty()) {
+            if (Patterns.EMAIL_ADDRESS.matcher(user).matches()) {
+                userSession.username = user
```

Comment:
You should also save the email in the UserSession object

Reply:
With "userSession.username = user" is not ok?